### DISCUSSION TITLE
Can students modify the exam configuration?

### DISCUSSION THREAD

**Discussion by Rapacioli** [SafeExamBrowser/seb-server#126] [Q&A]
I have a question, it is important to me. We are using seb server. To do this, we upload a file with the configuration in Moodle. My question is that I can modify that configuration and then take the exam with the new configuration. I understand that students can do the same and that seems like a serious flaw to me. Could you help me? From already thank you very much!
@dbuechel

**Comment by anhefti** [comment]
Hi,

Unfortunately, SEB Server Moodle integration is not compatible with the Moodles Deeper Integration of SEB. If you use SEB Server with Moodle we recommend to use the [Moodle SEB Server Plugin](https://seb-server.readthedocs.io/en/latest/lmssetup.html#install-seb-restriction-api-plugin). Please also read about how to setup and apply an Exam with SEB Server and SEB [here](https://seb-server.readthedocs.io/en/latest/configurations.html) and [here](https://seb-server.readthedocs.io/en/latest/exams.html). 

In short, when you are using SEB Server you only need a Connection Configuration for starting SEB while the Exam Configuration with the SEB Settings is automatically loaded by SEB when it connects to SEB Server. SEB always connect first to SEB Server and then to an Assessment Tool quiz page depending on student selection or preselected Connection Configuration. When you use Moodle SEB Server Plugin, SEB Server will automatically set "Use SEB Server" as restriction strategy and send SEB restriction details to Moodle. If you dont use the Moodle SEB Server Plugin, there will be no SEB Restriction on Moodle site while using SEB Server but without that is should function just like other Assessment Tool Setups an SEB Server.

Best regards

**Comment by Rapacioli** [comment]
Hi,
Thank you very much for your answer

I only have this exam configuration, I don't know if it is correctly setting:

[cid:2055a25a-0c2f-4bd4-bfab-974b50596638]

Then, for students to have access to the exam I must export the exam connection configuration:

[cid:88781132-72af-4f40-9cac-118094b079ff]

[cid:1c2d0b59-5d03-489f-962b-efd62053de6f]

And when I put this file in moodle student can download the file i am attaching

But that file can be modify by everyone

Can you please help me?

Thank you inadvance!

Best

Melina



Melina Rapacioli
Sistema Institucional de Educación a Distancia
Fundación Universitaria Dr. René Favaloro
Solís 453, Ciudad Autónoma de Buenos Aires (C1078AAI), Argentina
Tel.: +541140877231
***@***.******@***.***>; ***@***.******@***.***>
[cid:2953f036-0dca-451d-8982-f83f785602d9]
www.favaloro.edu.ar<http://www.favaloro.edu.ar/>
________________________________

**Reply by anhefti** [reply]
Hi,

Unfortunately, your links or screenshots are not displayed but steps seems to be right until with step:

"Then, for students to have access to the exam, I must export the exam connection configuration:"

Then you shall not upload your exam connection configuration to Moodle because Moodle Deeper Integration is not compatible with SEB Server configurations, but you can send this file directly to students or make it available by download on some other portal of yours. Students just need to open this File with SEB and SEB will automatically connect to SEB Server.

**Reply by Rapacioli** [reply]
Hi,
I am trying again to send the files:
![Connection Configruation](https://github.com/SafeExamBrowser/seb-server/assets/94466010/5cc4c3aa-9ac4-476a-8574-5e580c32e0e3)
![Export Connection Configuration](https://github.com/SafeExamBrowser/seb-server/assets/94466010/774320dc-9e51-4a0a-9ac2-a2e5857d0baa)
![Export Connection Configuration b](https://github.com/SafeExamBrowser/seb-server/assets/94466010/d7e89bee-73bc-47ec-bfb0-c8ca846ab2dc)
[MEDMINT124__Exmenes_con_SEB__2024_06_2024-6-3.zip](https://github.com/user-attachments/files/15534049/MEDMINT124__Exmenes_con_SEB__2024_06_2024-6-3.zip)

My question refers to students who want to commit fraud. When they are able to download the file, if they wanted to do fraud, before opening the file that guides them to SEB they could modify it and save it and then open it and they can use the modified configuration. This occurs even though the connection is set to start the exam. Even so, all the configuration is in the connection file that is exported. I would like to know if there is a way to export the connection without the exam settings being available, so they can't modify them.
THANK YOU!

**Comment by anhefti** [comment]
Hi, 

As I described, the connection configuration (or the exam connection configuration (same but with preselected exam)) **do not contain the SEB Settings of the Exam** but only the settings to securely connect to SEB Server. Then SEB loads the Exam Configuration with the SEB Settings for the Exam directly from SEB Server and reconfigure before connecting to Moodle.

If you want to secure the connection configuration too, you can either use a password encryption, where students then are asked for the password by SEB when loading the connection configuration, or with a certificate that is known by SEB Server as well as by SEB. But usually this is not necessary because a compromised connection configuration will probably not direct SEB to SEB Server and therefore a student will not get to the Exam either. 
And there are other fraud checks in place. If you use Moodle SEB Server Plugin, you have the SEB Restriction and are able to use the Browser Exam Key (BEK) To ensure SEB configuration and SEB app is not compromised. And on SEB Server you can use the ASK (App-Signature-Key) as described above to check SEB client integrity for an exam even with no special assessment tool integration.

Please read the documentation for SEB and SEB Server carefully, most of this is described there as well.

**Reply by anhefti** [reply]
Forgot to mention, that Students never should have access to SEB Server since SEB Server is the Admin-Tool. Some Exam Supporter should download and deliver the connection configuration file to Students in the way it is appropriate for you (by mail or put it on a file server with portal...).
